warp/Network/Wai/Handler/Warp/HTTP2/Receiver.hs

```diff
@@ -232,9 +232,12 @@ control FrameGoAway _ _ Context{controlQ} = do
 
 control FrameWindowUpdate header bs Context{connectionWindow} = do
     WindowUpdateFrame n <- guardIt $ decodeWindowUpdateFrame header bs
-    !w <- (n +) <$> atomically (readTVar connectionWindow)
+    !w <- atomically $ do
```

Comment:
Last time I checked (and I was sad) `TVar.modifyTVar` returns unit, unlike `modifyIORef`.

As per the strictness, the strict pattern binding '!w' will evaluate the thunk.